### ISSUE TITLE
KeyError: 'duration' when running both locally and on notepad

### ISSUE THREAD

**Issue by n71fs** [jquagga/tr-log-analyzer#33]
Im getting the following errors

Below is the error from the trlog.py running on my machine :
```
python3 trlog.py
Traceback (most recent call last):
  File "trlog.py", line 119, in <module>
    main()
  File "trlog.py", line 113, in main
    calldf = pandasconvert(calldict)
  File "trlog.py", line 71, in pandasconvert
    calldf.loc[calldf["duration"].notna(), "callclass"] = "standard"
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py", line 3761, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/indexes/range.py", line 349, in get_loc
    raise KeyError(key)
KeyError: 'duration'
```

And below here is the error from the Jupyter Lab notepad

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[1], line 57
     53 calldf = pd.DataFrame.from_dict(calldict, orient="index")
     54 # Technically this shouldn't be needed.  The dict construction _should_ set the class
     55 # but for some reason it skips setting standard.  It does set duration though so that part of
     56 # of the loop works.  This workaround sets the class to standard if there is a duration.
---> 57 calldf.loc[calldf["duration"].notna(), "callclass"] = "standard"
     59 # We're going to use ChanList.csv if we have it to convert decimal talkgroups to their
     60 # Alpha Longform.  While this could be in the original log line, we do it here to take care
     61 # of logs which might not have that enabled AND it allows us to see the number value of "unlisted" tg.
     62 try:

File /srv/conda/envs/notebook/lib/python3.10/site-packages/pandas/core/frame.py:4102, in DataFrame.__getitem__(self, key)
   4100 if self.columns.nlevels > 1:
   4101     return self._getitem_multilevel(key)
-> 4102 indexer = self.columns.get_loc(key)
   4103 if is_integer(indexer):
   4104     indexer = [indexer]

File /srv/conda/envs/notebook/lib/python3.10/site-packages/pandas/core/indexes/range.py:417, in RangeIndex.get_loc(self, key)
    415         raise KeyError(key) from err
    416 if isinstance(key, Hashable):
--> 417     raise KeyError(key)
    418 self._check_indexing_error(key)
    419 raise KeyError(key)

KeyError: 'duration'
```

See my tr.log.gz attached below as well.
[tr.log.gz](https://github.com/user-attachments/files/15964914/tr.log.gz)

**Comment by n71fs** [comment]
Here's the normal tr.log as well.

[tr.log](https://github.com/user-attachments/files/15964947/tr.log)

**Comment by jquagga** [comment]
Ok, I think I've sorted this one.  You're using the tag_id talkgroupDisplayFormat right?  The regexp's I created were for id_tag or well no talkgroup displayed at all.  I committed an update with the modified regexp - comment line 19 and uncomment line 24 and it should use / parse the file and spit out the details.  With that, I was able to get it to run on your sample file.  If that doesn't work, let me know!
